src/Conf.ml

```diff
@@ -899,7 +910,13 @@ let parse_line config ~from s =
 
 let rec read_conf_files conf ~dir ~parents =
   let dir' = Filename.dirname dir in
-  if (not (String.equal dir dir')) && Caml.Sys.file_exists dir then
+  let inside_root =
+    Option.value_map !root ~default:true
+      ~f:(String.is_substring ~substring:dir)
```

Comment:
I would be much more comfortable with converting dir and root to absolute paths (better would be to use Filename.realpath) and checking if calling dirname on dir zero or more times results in root, or using Filename.parts on both and checking if the root list is a prefix of the dir list. Basically, using filenames to do anything but open files is very fragile, and it would be better to try not to.

Reply:
There is this warning `Warning! this library assumes we are in a POSIX compliant OS. It will not work properly under windows.` in the Core.Filename documentation, so I think it is better to stick with the vanilla Filename lib.

Reply:
Well, treating paths as raw strings is significantly worse. Generally my position is that the consequence of windows failing to implement the established APIs should not be that everything is worse. It would be better to implement something robust, which someone who wants to work on windows can use as a specification for what windows-specific support is needed.

Also, there may be other libraries for filenames than Core, maybe even which have already resolved windows deficiencies.

Reply:
Could you have a look at using [fpath](https://github.com/dbuenzli/fpath) for this, in particular `normalize`. And unless realpath is used somewhere, the docs need to say that input files and project root need to be specified using the same symbolic or hard links.